File: src/cviz/ITimeline.java
package cviz;

import cviz.timeline.Trigger;
import se.svt.caspar.amcp.AmcpLayer;

import java.util.concurrent.CopyOnWriteArrayList;

public interface ITimeline {
    AmcpLayer getLayer(int layerId);

    void setLayerState(int layerId, LayerState state);
    LayerState getLayerState(int layerId);

    CopyOnWriteArrayList<Trigger> getActiveTriggers();

    String getTemplateData(String fieldName);
}

File: CViz/ITimeline.cs
using System;
using CViz.Timeline.Triggers;
using LibAtem.Net;
using StilSoft.CasparCG.AmcpClient;

namespace CViz
{
    interface ITimeline
    {
        AmcpConnection CasparClient { get; }
        AtemClient AtemClient { get; }

        int ChannelNumber { get; }

        void SetLayerState(int layerId, LayerState state);
        LayerState GetLayerState(int layerId);
        
        string GetParameterValue(string fieldName, bool escape);

        void AddTrigger(ITrigger t);
        void RemoveAllTriggers(Predicate<ITrigger> predicate);
    }
}